### PR TITLE
Edited dispatch plot with visualization

### LINKED ISSUE #261: [TASK] Update Dispatch Plotting Capabilities
Edited the dispatch plot script under HERON and changed it's output visualization from line plots to stackplots.
Created a pull request to address this issue.

----------------
For Change Control Board: Issue Review
----------------
This review should occur before any development is performed as a response to this issue.
- [x] 1. Is it tagged with a type: defect or task?
- [x] 2. Is it tagged with a priority: critical, normal or minor?
- [x] 3. If it will impact requirements or requirements tests, is it tagged with requirements?
- [x] 4. If it is a defect, can it cause wrong results for users? If so an email needs to be sent to the users.
- [x] 5. Is a rationale provided? (Such as explaining why the improvement is needed or why current code is wrong.)

-------
For Change Control Board: Issue Closure
-------
This review should occur when the issue is imminently going to be closed.
- [x] 1. If the issue is a defect, is the defect fixed?
- [x] 2. If the issue is a defect, is the defect tested for in the regression test system?  (If not explain why not.)
- [x] 3. If the issue can impact users, has an email to the users group been written (the email should specify if the defect impacts stable or master)?
- [x] 4. If the issue is a defect, does it impact the latest release branch? If yes, is there any issue tagged with release (create if needed)?
- [x] 5. If the issue is being closed without a pull request, has an explanation of why it is being closed been provided?

### PR DESCRIPTION
**Pull request by ghost**
--------
Pull Request Description
--------
##### What issue does this change request address?
#261

##### What are the significant changes in fu
dispatch_id1_y2021_c7

dispatch_id0_y2020_c2

### CODE REVIEW

File: src/DispatchPlot.py [hunk 1]
```diff
@@ -179,6 +200,50 @@ def plot_signal(self, fig, axes, df, sid, mstep, cid) -> None:
     self.raiseAMessage(f'Saved figure to "{signal_file_name}"')
     plt.clf()
 
+  def color_style(self, grp_vars):
+    """
+      @ In, grp_vars, Dict[str, List[str]], a dictionary mapping components to variables.
+      @ Out, cdict, Dict[str, str], contains color code for variables
```
Comment (PaulTalbot-INL) [inline src/DispatchPlot.py]: can we change the `cdict` output variable name to `colors` here? It looks like that's the name of the return variable within this method.

**Review by PaulTalbot-INL — COMMENTED** [review]
Only one recommendation for improvement.

**Comment by PaulTalbot-INL** [comment]
----------------
For Change Control Board: Change Request Review
----------------
The following review must be completed by an authorized member of the Change Control Board.
- [x] 1. Review all computer code.
- [x] 2. If any changes occur to the input syntax, there must be an accompanying change to the user manual and xsd schema. If the input syntax change deprecates existing input files, a conversion script needs to be added (see Conversion Scripts).
- [x] 3. Make sure the Python code and commenting standards are respected (camelBack, etc.) - See on the [wiki](https://github.com/idaholab/HERON/wiki/Code-Standards) for details.
- [x] 4. Automated Tests should pass.
- [x] 5. If significant functionality is added, there must be tests added to check this. Tests should cover all possible options.  Multiple short tests are preferred over one large tes.
- [x] 6. If the change modifies or adds a requirement or a requirement based test case, the Change Control Board's Chair or designee also needs to approve the change.  The requirements and the requirements test shall be in sync.
- [x] 7. The merge request must reference an issue.  If the issue is closed, the issue close checklist shall be done.
- [x] 8. If an analytic test is changed/added, the the analytic documentation must be updated/added.
- [x] 9. If any test used as a basis for documentation examples have been changed, the associated documentation must be reviewed and assured the text matches the example.

File: src/DispatchPlot.py (hunk 1 above)
Comment (ghost) [reply src/DispatchPlot.py]: Sounds good. I'll change the variable name and push it to repo.

**Comment by dylanjm** [comment]
Hi @yangx789 can we also make it so that white is not used as a legend color? For example, the Additional NPP variable in the plots above is really hard to see. 

Also, can we set up the secondary y-axis to track the values of the storage levels in the problem too?

File: src/DispatchPlot.py [hunk 2]
```diff
@@ -104,7 +106,16 @@ def _group_by(iterable: List[str], idx: int) -> Dict[str, List[str]]:
         gr[key] = [var]
     return gr
 
-  def plot_component(self, fig, axes, df, grp_vars, comp_idx, sid, mstep, cid) -> None:
+  def align_yaxis(ax1, v1, ax2, v2):
+    """adjust ax2 ylimit so that v2 in ax2 is aligned to v1 in ax1"""
```
Comment (dylanjm) [inline src/DispatchPlot.py]: please format the docstrings of this method like the others.

**Review by dylanjm — CHANGES_REQUESTED** [review]

File: src/DispatchPlot.py (hunk 2 above)
Comment (ghost) [reply src/DispatchPlot.py]: Sorry, that was a test function that I forgot to delete. I'll remove that part and push the change again.

**Review by dylanjm — APPROVED** [review]
Changes look good to me and is ready to merge.

**Pull request merged**